test/unit/lib/offline-api/validation/payload-validation-resource-links.spec.ts

```diff
@@ -194,35 +194,133 @@ describe('payload validation (dependencies)', function () {
           .name('Main Course')
           .type('ResourceLink')
           .allowedResources([
-            {
-              type: 'Contentful:Entry',
-              source: 'crn:contentful:::content:spaces/cooking-space-1',
-              contentTypes: []
-            },
             {
               type: 'Contentful:Asset',
               source: 'crn:contentful:::content:spaces/cooking-space-2',
+              contentTypes: ['cookingMethod']
+            }
+          ])
+      }, existingCts)
+
+      expect(errors).to.eql([
+        [
+          {
+            type: 'InvalidPayload',
+            message:
+              'Allowed resource at index 0 on the field "mainCourse" has an invalid property: "type" must be [Contentful:Entry].'
+          }
+        ]
+      ])
+    })
+
+    it('returns errors when the allowedResources of the Contentful: Provider contains additional fields', async function () {
+      const existingCts = []
+      const errors = await validateBatches(function (migration) {
+        const lunch = migration.createContentType('lunch').name('Lunch').description('A Lunch')
+
+        lunch
+          .createField('mainCourse')
+          .name('Main Course')
+          .type('ResourceLink')
+          .allowedResources([
+            {
+              type: 'Contentful:Entry',
+              source: 'crn:contentful:::content:spaces/cooking-space-1',
               contentTypes: ['cookingMethod'],
               unexpectedProperty: 'someValue'
             }
           ])
       }, existingCts)
 
+      expect(errors).to.eql([
+        [
+          {
+            message:
+              'Allowed resource at index 0 on the field "mainCourse" has an invalid property: "unexpectedProperty" is not allowed.',
+            type: 'InvalidPayload'
+          }
+        ]
+      ])
+    })
+
+    it('returns errors when the contentTypes array is empty', async function () {
+      const existingCts = []
+      const errors = await validateBatches(function (migration) {
+        const lunch = migration.createContentType('lunch').name('Lunch').description('A Lunch')
+
+        lunch
+          .createField('mainCourse')
+          .name('Main Course')
+          .type('ResourceLink')
+          .allowedResources([
+            {
+              type: 'Contentful:Entry',
+              source: 'crn:contentful:::content:spaces/cooking-space-1',
+              contentTypes: []
+            }
+          ])
+      }, existingCts)
+
       expect(errors).to.eql([
         [
           {
             type: 'InvalidPayload',
             message:
               'Allowed resource at index 0 on the field "mainCourse" has an invalid property: "contentTypes" must contain at least 1 items.'
-          },
+          }
+        ]
+      ])
+    })
+
+    it('should return an error when the type value is referencing a invalid type', async function () {
```

Comment:
nitpick: use "returns" or "should return" consistently

Reply:
Yea, it's duplicated code vs my own code. I usually prefer `it('should return ..')` while the pre-existing code uses `it('returns an error')`.

I'll change it to match the wording already existent in the repo but don't think it's something we should strive to write consistently as long as the test name is clear

Reply:
Looking at the repo there are [6 more tests](https://github.com/contentful/contentful-migration/blob/abc0a9d94aee4dba19ef4a6016ce0dbcb3ffcf1e/test/unit/lib/offline-api/validation/payload-validation-rich-text-resource-link-embeds.spec.ts#L148) that use the `'should return ..'` syntax, all written by me 🙈 I did not change them as it's not in scope of this PR.